File: go/008/main.go
/*
 * ProjectEuler 008
 * Go / main.go
 * copyright (c) 2015 Susisu
 *
 * Go 1.4.1
 * `go build -o bin/main main.go`
 */


package main

import (
    "fmt"
    "log"
    "os"
)

func main() {
    l := 1000

    file, err := os.Open("./digits.txt")
    if err != nil {
        log.Fatal(err)
    }
    digits := make([]byte, l)
    _, err = file.Read(digits)
    if err != nil {
        log.Fatal(err)
    }
    file.Close()

    n := 13
    max := 0
    for i := 0; i <= l - n; i++ {
        p := 1
        for j := 0; j < n; j++ {
            p *= digitToInt(digits[i + j])
        }
        if p > max {
            max = p
        }
    }
    fmt.Printf("%d\n", max)
}

func digitToInt(d byte) int {
    n := int(d) - 48
    if n < 0 || 9 < n {
        return 0
    } else {
        return n
    }
}

File: lua/008/main.lua
--[[
    ProjectEuler 008
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    local file = io.open("./digits.txt")
    local digits = file:read("*l")
    local len = string.len(digits)
    local n = 13
    local max = 0
    for i = 1, len - n + 1 do
        local p = 1
        for j = i, i + n - 1 do
            p = p * tonumber(string.sub(digits, j, j))
        end
        if p > max then
            max = p
        end
    end
    print(max)
end

main()
